PopPUNK/lineages.py

```diff
@@ -0,0 +1,470 @@
+            strain_rlist,
+            overall_lineage[strain],
+            output_format = 'phandango',
+            epiCsv = None,
+            suffix = '_Lineage')
+        genomeNetwork = indivNetworks[min(rank_list)]
+        # Save model
+        model.save()
+
+    # Print combined strain and lineage clustering
+    print_overall_clustering(overall_lineage,args.output + '.csv',rlist)
+
+    # Write scheme to file
+    with open(args.db_scheme, 'wb') as pickle_file:
+        pickle.dump([args.db,
+                      rlist,
+                      args.model_dir,
+                      clustering_file,
+                      args.clustering_col_name,
+                      distances,
+                      kmers,
+                      sketch_sizes,
+                      codon_phased,
+                      max_search_depth,
+                      rank_list,
+                      args.use_accessory,
+                      args.min_count,
+                      args.count_unique_distances,
+                      args.reciprocal_only,
+                      args.strand_preserved,
+                      args.core,
+                      args.accessory,
+                      lineage_dbs],
+                    pickle_file)
+
+    return 0
+
+def query_db(args):
+    
+    # Read querying scheme
+    with open(args.db_scheme, 'rb') as pickle_file:
+        ref_db, rlist, model_dir, clustering_file, args.clustering_col_name, distances, \
+        kmers, sketch_sizes, codon_phased, max_search_depth, rank_list, use_accessory, min_count, \
+        count_unique_distances, reciprocal_only, strand_preserved, core, accessory, lineage_dbs = \
+          pickle.load(pickle_file)
+
+    dbFuncs = setupDBFuncs(args)
+
+    # Define clustering files
+    previous_clustering_file = os.path.join(model_dir,os.path.basename(model_dir) + '_clusters.csv')
+    external_clustering = None
+    if clustering_file != previous_clustering_file:
+        external_clustering = clustering_file
+
+    # Store clustering
+    overall_lineage = {}
+
+    # Ignore QC at the moment
+    qc_dict = {'run_qc': False, 'type_isolate': None }
+
+    # Check output file
+    if args.output is None:
+        sys.stderr.write("Need an output file name\n")
+        sys.exit(1)
+
+    # Set up database
+    createDatabaseDir = dbFuncs['createDatabaseDir']
+    constructDatabase = dbFuncs['constructDatabase']
+    readDBParams = dbFuncs['readDBParams']
+
+    if ref_db == args.output:
+        sys.stderr.write("--output and --ref-db must be different to "
+                         "prevent overwrite.\n")
+        sys.exit(1)
+
+    # construct database
+    createDatabaseDir(args.output, kmers)
+    qNames = constructDatabase(args.query,
+                                kmers,
+                                sketch_sizes,
+                                args.output,
+                                args.threads,
+                                True, # overwrite - probably OK?
```

Comment:
make a kwarg

Reply:
Will fix this when we've decided on workflow